### PR TITLE
[2.0.x] Port 482, 542, 595, 674, 536 and part of the 281

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

**Comment by dmvolod** [comment]
@rigazilla PR is ready for review

**Review by rigazilla — COMMENTED** [review]
Seems good to me.
@dmvolod are existing CRs compatible with new CRD constraints? Have you tested upgrades from older version with clusters deployed?

**Comment by dmvolod** [comment]
> @dmvolod are existing CRs compatible with new CRD constraints? Have you tested upgrades from older version with clusters deployed?

Not yet migration was tested. Will try to do tomorrow

**Comment by dmvolod** [comment]
Thanks for pointing me with Upgrade test, @rigazilla 
Found another update case regression. Need to forwardport it again to `master`
Will push new version soon

**Comment by dmvolod** [comment]
@rigazilla the PR is ready for review. Seems to be extra stable with ported `controllerutil.CreateOrPatch` function

**Comment by ryanemerson** [comment]
Is there a Issue to track this work? I don't why the backports/fixes are required ...

File: pkg/controller/infinispan/util/controllerutil.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
+package util
```
Comment (ryanemerson) [inline pkg/controller/infinispan/util/controllerutil.go:1]: As this is contained within the `util` package, I think we can simply call the file `pkg/controller/infinispan/util/controller.go`

Comment (dmvolod) [reply pkg/controller/infinispan/util/controllerutil.go]: Fixed

**Comment by dmvolod** [comment]
> Is there a Issue to track this work? I don't why the backports/fixes are required ...

Oh, sorry, @ryanemerson missed your question
#711 - issue with upgrade
#536 -adds additional value for TravisCI test
#712 #714 - adds additional stability to updates `status` and `spec`
Backport upgrade-test-local and upgrade-test needs for upgrade validation as `2.0.x` still in active support

**Review by rigazilla — APPROVED** [review]

**Comment by ryanemerson** [comment]
Thanks @dmvolod

**Pull request merged**